kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/CandidateDetailProviderImpl.java

```diff
@@ -0,0 +1,141 @@
+package org.sakaiproject.user.impl;
```

Comment:
I was expecting these providers to be in the /providers folder as although the API needs to be in the kernel, the actual implementation of the API (like UserDirectoryProvider) can just be in the providers.